plugins/test-management-resources/src/components/test-case/CreateTestCase.svelte

```diff
@@ -0,0 +1,192 @@
+<!--
+// Copyright © 2024 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte'
+  import { Attachment } from '@hcengineering/attachment'
+  import { AttachmentPresenter, AttachmentStyledBox } from '@hcengineering/attachment-resources'
+  import { TestCase, TestProject, TestSuite, TestCaseStatus } from '@hcengineering/test-management'
+  import core, { fillDefaults, generateId, makeCollaborativeDoc, Ref, TxOperations, Data } from '@hcengineering/core'
+  import { ObjectBox } from '@hcengineering/view-resources'
+  import { Card, SpaceSelector, getClient, updateMarkup } from '@hcengineering/presentation'
+  import { EmptyMarkup } from '@hcengineering/text'
+  import { Button, createFocusManager, EditBox, FocusHandler, IconAttachment, getLocation } from '@hcengineering/ui'
+
+  import StatusEditor from './StatusEditor.svelte'
+  import AssigneeEditor from './AssigneeEditor.svelte'
+  import ProjectPresenter from '../project/ProjectPresenter.svelte'
+  import testManagement from '../../plugin'
+
+  export let onCreate: ((orgId: Ref<TestCase>, client: TxOperations) => Promise<void>) | undefined = undefined
+
+  export function canClose (): boolean {
+    return object.name === ''
+  }
+
+  export let space: Ref<TestProject>
+
+  export let testSuiteId: Ref<TestSuite> | undefined
+
+  testSuiteId = testSuiteId ?? (getLocation()?.query?.suite as Ref<TestSuite>)
+  const id: Ref<TestCase> = generateId()
+
+  const object: Data<TestCase> = {
+    name: '',
+    description: makeCollaborativeDoc(id, 'description'),
+    status: TestCaseStatus.Draft,
+    assignee: null,
+    attachments: 0,
+    suite: testSuiteId
+  } as unknown as TestCase
+
+  let _space = space
+
+  const dispatch = createEventDispatcher()
+  const client = getClient()
+  const hierarchy = client.getHierarchy()
+
+  let description = EmptyMarkup
+
+  fillDefaults(hierarchy, object, testManagement.class.TestCase)
+
+  async function createTestCase (): Promise<void> {
+    const op = client.apply()
+    await updateMarkup(object.description, { description })
+    await op.createDoc(testManagement.class.TestCase, _space, object, id)
```

Comment:
If TestCase is an AttachedDoc, then you should use `addCollection` method.

Reply:
Thank you, fixed